clients/src/main/java/org/apache/kafka/clients/ClientDnsLookup.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum ClientDnsLookup {
+
+    RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY("resolve.canonical.bootstrap.servers.only"),
```

Comment:
@lepolac @mimaison @edoardocomar What was the conclusion on the discussion regarding naming of these config value? Did we decide to go with underscores or dots as separator?

Reply:
My bias would be dots given the number of parameters using dots.
https://kafka.apache.org/11/documentation.html#consumerconfigs

Reply:
A newish enum-based config like `isolation.level` in ConsumerConfig uses underscores for values, and as @mimaison noted we can leverage that in the `fromString`

So, my bias is for underscores but happy to change to dots ...

Reply:
@lepolac Just to make sure we are on the same page, the config name will remain dot-separated (i.e. `client.dns.lookup`). But the enum values can be underscore separated with the same name as the enum in lower case (i.e. `resolve_canonical_bootstrap_servers_only`). Not sure we do dot-separation for enum values in other configs, but may be we do.

Reply:
Config names use dots for separators but most config values use underscores. For example: `sasl_ssl`, `read_uncommitted`, `at_least_once`

Reply:
Oh right, yes sorry I was confusing config name and value. Indeed it looks like "_" is more used, will modify accordingly.